src/js/report_form.js

```diff
@@ -41,6 +52,60 @@
     }
   }
 
+  /**
+   * Converts the specified time to the timezone provided
+   *
+   * @param {string} time string representing HH:mm string
+   * @param {string} tz which timezone to convert to
+   */
+  function convertToTimezone(time, tz) {
+    if (time === "") {
+      return "";
+    }
+    let dateTimeObj = moment.utc(`${defaultDate} ${time}`);
+    dateTimeObj.tz(tz);
+    return dateTimeObj;
+  }
+  /**
+   * Gets current timezone from local storage
+   * @returns {datetime} either the browser
+   *   timezone or manually selected timezone
+   */
+  function getSelectedTimezone() {
+    const manualTz = localStorage.getItem("chosen-timezone");
+    const selectedTz = localStorage.getItem("selected-timezone");
+    return selectedTz || manualTz;
+  }
+  /**
+   * Gets current timezone from local storage
+   * @param {scheduleType} type of schedule so we know whether
+   *   to update the weekdayinput
+   */
+  function convertTimesToLocalTimezone(scheduleType) {
+    scheduleTimezoneInput.value = getSelectedTimezone();
+    let convertedTime = convertToTimezone(
+      scheduleTimeInput.value,
+      scheduleTimezoneInput.value
```

Comment:
i'm using the setting from the hidden timezone input so i can pass it to the backend in the form data. im avoiding calling the function twice because the value is already stored